cpp/examples/tpch/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) 2020-2024, NVIDIA CORPORATION.
+
+cmake_minimum_required(VERSION 3.26.4)
+
+include(../set_cuda_architecture.cmake)
+
+rapids_cuda_init_architectures(tpch_example)
+rapids_cuda_set_architectures(RAPIDS)
+
+project(
+  tpch_example
+  VERSION 0.0.1
+  LANGUAGES CXX CUDA
+)
+
+include(../fetch_dependencies.cmake)
+
+add_executable(tpch_q1 q1.cpp)
+target_link_libraries(tpch_q1 PRIVATE cudf::cudf)
+target_compile_features(tpch_q1 PRIVATE cxx_std_17)
+
+add_executable(tpch_q5 q5.cpp)
+target_link_libraries(tpch_q5 PRIVATE cudf::cudf)
+target_compile_features(tpch_q5 PRIVATE cxx_std_17)
+
+add_executable(tpch_q6 q6.cpp)
+target_link_libraries(tpch_q6 PRIVATE cudf::cudf)
+target_compile_features(tpch_q6 PRIVATE cxx_std_17)
+
+add_executable(tpch_q9 q9.cpp)
+target_link_libraries(tpch_q9 PRIVATE cudf::cudf)
+target_compile_features(tpch_q9 PRIVATE cxx_std_17)
```

Comment:
I think it may be a good idea to add one or more small data files to test this example as a part of CI as well.

To do that, please commit the data file (say cudf/cpp/examples/tpch/tpchXX.parquet) and install one or more example binaries with the data file like this here:
```cmake
install(TARGETS tpch_qX DESTINATION bin/examples/libcudf)
install(FILES ${CMAKE_CURRENT_LIST_DIR}/tpchXX.parquet DESTINATION bin/examples/libcudf)
```

and then in [run_cudf_examples.sh](https://github.com/rapidsai/cudf/blob/branch-24.08/ci/run_cudf_examples.sh) add something like:

```bash
compute-sanitizer --tool memcheck tpch_qXX tpchXX.parquet [any other args?]
```

Reply:
I think we can skip this example in CI for now!